packages/node/src/utils/subProcessRunner.ts

```diff
@@ -0,0 +1,110 @@
+import { ChildProcess, fork } from 'child_process';
+import { PromiseQueue } from './promise-queue';
+import { NodeBuildEvent } from '../executors/build/build.impl';
+import { NodeExecuteBuilderOptions } from '../utils/types';
+import { InspectType } from '@nrwl/node/src/executors/execute/execute.impl';
+import { promisify } from 'util';
+import * as treeKill from 'tree-kill';
+import { logger } from '@nrwl/devkit';
+
+export type SubProcessEvent =
+  | {
+      exitCode: number;
+    }
+  | {
+      message: string;
+    }
+  | {
+      error: unknown;
+    };
+
+export class SubProcessRunner {
```

Comment:
Happy to write tests for this, awaiting feedback for where this should be in the project structure. It seems like it belongs colocated to `executor.impl.ts`, but I didn't see other examples of doing stuff like that.

The executor.impl tests cover the integration of this in full, but it felt big enough to split into a separate file. Let me know what you think!